homeassistant/components/ozw/services.py

```diff
@@ -37,6 +43,93 @@ def async_register(self):
             ),
         )
 
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_SET_CONFIG_PARAMETER,
+            self.async_set_config_parameter,
+            schema=vol.Schema(
+                {
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                    vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
+                    vol.Required(const.ATTR_CONFIG_PARAMETER): vol.Coerce(int),
+                    vol.Required(const.ATTR_CONFIG_VALUE): vol.Any(
+                        vol.Coerce(int), cv.string
+                    ),
+                }
+            ),
+        )
+
+    @callback
+    def async_set_config_parameter(self, service):
+        """Set a config parameter to a node."""
+        instance_id = service.data[const.ATTR_INSTANCE_ID]
+        node_id = service.data[const.ATTR_NODE_ID]
+        param = service.data[const.ATTR_CONFIG_PARAMETER]
+        selection = service.data[const.ATTR_CONFIG_VALUE]
+        payload = None
+
+        node = self._manager.get_instance(instance_id).get_node(node_id).values()
+
+        for value in node:
+            if (
+                value.command_class != CommandClass.CONFIGURATION
+                or value.index != param
+            ):
+                continue
+            if value.type == ValueType.BOOL:
+                payload = selection == "True"
+
+            if value.type == ValueType.LIST:
+                # accept either string from the list value OR the int value
+                if isinstance(selection, int):
+                    if selection > value.max or selection < value.min:
+                        _LOGGER.error(
+                            "Value %s out of range for parameter %s (Min: %s Max: %s)",
+                            selection,
+                            param,
+                            value.min,
+                            value.max,
+                        )
+                        break
+                    payload = int(selection)
+
+                # iterate list labels to get value
+                for selected in value.value["List"]:
+                    if selected["Label"] != selection:
+                        continue
+                    payload = int(selected["Value"])
+
+                if payload is None:
+                    _LOGGER.error(
+                        "Invalid value %s for parameter %s", selection, param,
+                    )
+                    break
+
+            if value.type == ValueType.BUTTON:
+                # Unsupported at this time
+                _LOGGER.info("Button type not supported yet")
+                return
+            if value.type == ValueType.INT:
```

Comment:
What about BYTE and SHORT? Those also have min and max values that can be checked in the same way.
```json
{
    "Label": "Luminance Reporting Threshold",
    "Value": 10,
    "Units": "%",
    "ValueSet": false,
    "ValuePolled": false,
    "ChangeVerified": false,
    "Min": 0,
    "Max": 50,
    "Type": "Byte",
    "Instance": 1,
    "CommandClass": "COMMAND_CLASS_CONFIGURATION",
    "Index": 4,
    "Node": 2,
    "Genre": "Config",
    "Help": "Threshold change in luminance to induce an automatic report. 0 is disabled. 5 to 50 ( Set up from 5%-50%)",
    "ValueIDKey": 1125899950620689,
    "ReadOnly": false,
    "WriteOnly": false,
    "Event": "valueAdded",
    "TimeStamp": 1594103629
}
```

Reply:
Byte, Short those get converted to int, I'll add em in